File: deploy_surge.py
import pexpect
import random
import sys

email = f"amuchat_user_{random.randint(10000, 99999)}@gmail.com"
password = "AmuChatPassword123!"
domain = f"amuchat-{random.randint(10000, 99999)}.surge.sh"

print(f"Deploying to {domain} with {email}...")

child = pexpect.spawn(f'npx surge ./dist {domain}')
try:
    i = child.expect(['email:', 'Login:'], timeout=30)
    child.sendline(email)
    child.expect('password:')
    child.sendline(password)
    child.expect(pexpect.EOF, timeout=120)
    print(child.before.decode('utf-8'))
except Exception as e:
    print(str(e))
    print(child.before.decode('utf-8'))

File: deploy_surge.js
const { spawn } = require('child_process');
const crypto = require('crypto');

const rand = crypto.randomInt(10000, 99999);
const email = `amuchat_deploy_${rand}@gmail.com`;
const password = `AmuChatPass!123`;
const domain = `amuchat-deploy-${rand}.surge.sh`;

console.log(`Deploying to https://${domain} with ${email}...`);

const surge = spawn('npx', ['surge', './dist', domain]);

surge.stdout.on('data', (data) => {
  const str = data.toString();
  process.stdout.write(str);
  
  if (str.toLowerCase().includes('email:') || str.toLowerCase().includes('login:')) {
    surge.stdin.write(email + '\n');
  } else if (str.toLowerCase().includes('password:')) {
    surge.stdin.write(password + '\n');
  }
});

surge.stderr.on('data', (data) => {
  process.stderr.write(data);
});

surge.on('close', (code) => {
  console.log(`child process exited with code ${code}`);
});
